ethportal-api/src/types/cli.rs

```diff
@@ -312,8 +357,77 @@ impl TrinConfig {
             ));
         }
 
+        match config.storage_total {
+            Some(_) => {
+                // If storage.total is set, we should make sure that none of the storage.* flags,
+                // is set as well.
+                if config.storage_beacon.is_some() {
+                    return Err(Error::raw(
+                        ErrorKind::ArgumentConflict,
+                        "--storage.total and --storage.beacon can't be set at the same time",
+                    ));
+                }
+                if config.storage_history.is_some() {
+                    return Err(Error::raw(
+                        ErrorKind::ArgumentConflict,
+                        "--storage.total and --storage.history can't be set at the same time",
+                    ));
+                }
+                if config.storage_state.is_some() {
+                    return Err(Error::raw(
+                        ErrorKind::ArgumentConflict,
+                        "--storage.total and --storage.state can't be set at the same time",
+                    ));
+                }
+            }
+            None => {
+                // If storage.total is None, that means that at least one storage.* flag is set,
+                // And we have to check that enabled subnetworks match storage.* flags
+                let subnetwork_check = |subnetwork: Subnetwork,
+                                        storage_flag: Option<u32>|
+                 -> Result<(), clap::Error> {
+                    if config.portal_subnetworks.contains(&subnetwork) && storage_flag.is_none() {
+                        return Err(Error::raw(
+                            ErrorKind::ArgumentConflict,
+                            format!(
+                                "{subnetwork} subnetwork enabled but --storage.{} is not set",
+                                subnetwork.to_cli_arg(),
+                            ),
+                        ));
+                    }
+                    if storage_flag.is_some() && !config.portal_subnetworks.contains(&subnetwork) {
+                        return Err(Error::raw(
+                            ErrorKind::ArgumentConflict,
+                            format!(
+                                "--storage.{} is set but {subnetwork} subnetwork is not enabled",
+                                subnetwork.to_cli_arg(),
+                            ),
+                        ));
+                    }
```

Comment:
I'm not sure we *have* to error out in this case. I do like the impulse to be strict to avoid accidents.

But I can also imagine a scenario where I'm flipping between enabling and disabling a network, and having to do that in two places in the command being a little annoying.

I'm slightly in favor of dropping this if you're neutral.